packages/common/src/audius-query/hooks/usePaginatedQuery.ts

```diff
@@ -64,28 +65,52 @@ export const useAllPaginatedQuery = <
     offset: page * pageSize
   } as ArgsType
   const result = useQueryHook(args, queryHookOptions)
-  useEffect(() => {
-    if (result.status !== Status.SUCCESS) return
-    setAllData((allData) => [...allData, ...result.data])
-  }, [result.status, result.data])
 
   useCustomCompareEffect(
     () => {
       setAllData([])
+      setLoadingMore(false)
     },
     [baseArgs],
-    isEqual
+    (prevDeps, nextDeps) => {
+      return isEqual(prevDeps, nextDeps)
+    }
+  )
+
+  useCustomCompareEffect(
+    () => {
+      if (!statusIsNotFinalized(result.status)) {
+        setLoadingMore(false)
+      }
+      if (result.status !== Status.SUCCESS) return
+      setAllData((allData) => [...allData, ...result.data])
+    },
+    [result.status, args],
+    (prevDeps, nextDeps) => {
+      return isEqual(prevDeps, nextDeps)
+    }
   )
 
+  const hasMore =
```

Comment:
nit for readability

```
const notError = result.status !== Status.ERROR
const idleAndNoData = result.status === Status.IDLE && allData.length === 0
const hasNotFetched = !result.data && result.status !== Status.SUCCESS
const fetchedFullPreviousPage = result.data?.length === pageSize

const hasMore = notError && (idleAndNoData || hasNotFetched || fetchedFullPreviousPage)
```

Not quite sure if I understand the `idleAndNoData` case though

Reply:
ty great suggestion! idle + no data is just the initial state before anything loads